huaweicloud/services/dcs/data_source_huaweicloud_dcs_maintainwindow.go

```diff
@@ -45,20 +47,20 @@ func DataSourceDcsMaintainWindowV1() *schema.Resource {
 	}
 }
 
-func dataSourceDcsMaintainWindowV1Read(d *schema.ResourceData, meta interface{}) error {
+func dataSourceDcsMaintainWindowRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*config.Config)
-	dcsV1Client, err := config.DcsV1Client(GetRegion(d, config))
+	dcsV2Client, err := config.DcsV2Client(config.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("Error creating dcs key client: %s", err)
+		return fmtp.DiagErrorf("Error creating dcs key client: %s", err)
 	}
 
-	v, err := maintainwindows.Get(dcsV1Client).Extract()
+	v, err := maintainwindows.Get(dcsV2Client).Extract()
 	if err != nil {
-		return err
+		return diag.FromErr(err)
 	}
 
 	maintainWindows := v.MaintainWindows
-	var filteredMVs []maintainwindows.MaintainWindow
+	filteredMVs := make([]maintainwindows.MaintainWindow, 0, len(maintainWindows))
```

Comment:
are you interesting to try `utils.FliterSliceWithField`?

Reply:
I tried it, may be we need to make some modifications to it.
```
utils.FliterSliceWithField(maintainWindows, map[string]interface{}{
		"seq":     d.Get("seq").(int),
		"begin":   d.Get("begin").(string),
		"end":     d.Get("end").(string),
		"default": d.Get("default").(bool),
	})
```

the value of the key may be empty, FillerSliceWithField cannot ignore them